<!DOCTYPE html><html lang="zh-CN"><head><meta http-equiv="content-type" content="text/html; charset=utf-8"><meta content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" name="viewport"><meta content="yes" name="apple-mobile-web-app-capable"><meta content="black-translucent" name="apple-mobile-web-app-status-bar-style"><meta content="telephone=no" name="format-detection"><meta name="description" content="Some resources for learning Bioinformatics and Genome analysis"><title>BLAST+使用教程 | LIGENE</title><link rel="stylesheet" type="text/css" href="/css/style.css?v=0.0.0"><link rel="stylesheet" type="text/css" href="//cdn.bootcss.com/normalize/7.0.0/normalize.min.css"><link rel="stylesheet" type="text/css" href="//cdn.bootcss.com/pure/1.0.0/pure-min.css"><link rel="stylesheet" type="text/css" href="//cdn.bootcss.com/pure/1.0.0/grids-responsive-min.css"><link rel="stylesheet" href="//cdn.bootcss.com/font-awesome/4.7.0/css/font-awesome.min.css"><script type="text/javascript" src="//cdn.bootcss.com/jquery/3.2.1/jquery.min.js"></script><link rel="Shortcut Icon" type="image/x-icon" href="/favicon.ico"><link rel="apple-touch-icon" href="/apple-touch-icon.png"><link rel="apple-touch-icon-precomposed" href="/apple-touch-icon.png"><link rel="alternate" type="application/atom+xml" href="/atom.xml"></head><body><div class="body_container"><div id="header"><div class="site-name"><h1 class="hidden">BLAST+使用教程</h1><a id="logo" href="/.">LIGENE</a><p class="description">Notes of Bioinformatics and Genomics</p></div><div id="nav-menu"><a class="current" href="/."><i class="fa fa-home"> 首页</i></a><a href="/archives/"><i class="fa fa-archive"> 归档</i></a><a href="/bigbook/"><i class="fa fa-book"> BiG教程</i></a><a href="/about/"><i class="fa fa-user"> 关于</i></a></div></div><div class="pure-g" id="layout"><div class="pure-u-1 pure-u-md-3-4"><div class="content_container"><div class="post"><h1 class="post-title">BLAST+使用教程</h1><div class="post-meta">Jul 5, 2017<span> | </span><span class="category"><a href="/categories/Bioinformatics/">Bioinformatics</a></span></div><div class="post-content"><p>Linux下最新版BLAST+的本地化安装与基本使用用法。<br>BLAST的更详细使用方法可参考BLAST手册《BLAST Command Line Applications User Manual》，网址：<a href="http://www.ncbi.nlm.nih.gov/books/NBK279690/" target="_blank" rel="noopener">http://www.ncbi.nlm.nih.gov/books/NBK279690/</a><br><a id="more"></a></p>
<h2 id="1-下载"><a href="#1-下载" class="headerlink" title="1. 下载"></a>1. 下载</h2><p>从NCBI网址下载最新版BLAST程序：ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/<br>根据自己的操作系统选择相应的BLAST可执行程序，如Ubuntu 12.04(64位)系统就下载：<br>$wget ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/ncbi-blast-2.6.0+-x64-linux.tar.gz</p>
<h2 id="2-安装"><a href="#2-安装" class="headerlink" title="2. 安装"></a>2. 安装</h2><ul>
<li>解压缩：<br>$tar -zxvf ncbi-blast-2.6.0+-x64-linux.tar.gz</li>
<li>移动到本地安装目录（不带版本号，避免因升级而修改配置文件）<br>$sudo mv ncbi-blast-2.6.0+/  /opt/blast</li>
<li>将BLAST+可执行程序所在目录（bin）的绝对路径加入到环境变量$PATH中，方便通过程序名直接调用。<br>编辑~/.bashrc文件，在最后加入bin目录<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">echo &quot;export PATH=/opt/blast/bin:\$PATH&quot; &gt;&gt; ~/.bashrc    </span><br><span class="line">source ~/.bashrc  #让配置生效</span><br><span class="line">blastn -version  #显示版本</span><br></pre></td></tr></table></figure>
</li>
</ul>
<p>如果显示版本正确，安装成功。</p>
<h2 id="3-配置本地BLAST库"><a href="#3-配置本地BLAST库" class="headerlink" title="3.配置本地BLAST库"></a>3.配置本地BLAST库</h2><ul>
<li><p>安装目录/opt/blast下新建一个文件夹，命名为db 。<br>$mkdir /opt/blast/blastdb</p>
</li>
<li><p>在用户home目录下新建一个文件.ncbirc（文件名是以点号开头）。<br>$vi ~/.ncbirc<br>在文件中添加内容：<br>[BLAST]<br>BLASTDB=/opt/blast/blastdb</p>
</li>
</ul>
<p>配置好后，以后做BLAST选择库的时候就可以只输入名称（比如nr），不用输入绝对路径了。</p>
<h2 id="4-基本用法"><a href="#4-基本用法" class="headerlink" title="4.基本用法"></a>4.基本用法</h2><p>BLAST程序执行前要先准备查找目标序列(Subject)的数据库，BLAST程序需要有特定格式数据库才能进行搜索，所以要先格式化序列数据库；也可以直接从NCBI下载已经格式化好的数据库，如NT, NR数据库等。然后根据查询序列(Query)为蛋白质或核酸序列选择相应BLAST程序，如blastp或blastn等搜索相似序列。</p>
<h3 id="4-1-格式化序列数据库"><a href="#4-1-格式化序列数据库" class="headerlink" title="4.1 格式化序列数据库"></a>4.1 格式化序列数据库</h3><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">$makeblastdb -in db.fasta -dbtype nucl -parse_seqids -out dbname</span><br><span class="line">参数说明:</span><br><span class="line">-in：待格式化的序列文件</span><br><span class="line">-dbtype：数据库类型，prot或nucl</span><br><span class="line">-out：数据库名</span><br></pre></td></tr></table></figure>
<p>如果该库需要经常使用，可将库文件移到前面配置的库文件的目录，今后在其它目录运行blast的时候，便可直接输入库名（不用输入绝对路径），直接使用。<br>$mv dbname.* /opt/blast/blastdb</p>
<h3 id="4-2-运行BLAST程序"><a href="#4-2-运行BLAST程序" class="headerlink" title="4.2 运行BLAST程序"></a>4.2 运行BLAST程序</h3><ul>
<li>以核酸序列比对核酸数据库（blastn）为例：<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line">$blastn -query seq.fasta -out seq.blast -db dbname -outfmt 6 -evalue 1e-5 -num_descriptions 10 -num_threads 8</span><br><span class="line">参数说明:</span><br><span class="line">	-query： 输入文件路径及文件名</span><br><span class="line">	-out：输出文件路径及文件名</span><br><span class="line">	-db：格式化了的数据库路径及数据库名</span><br><span class="line">	-outfmt：输出文件格式，总共有17种格式，6是tabular格式（对应BLAST的m8格式），最新版BLAST+的17是sam格式。</span><br><span class="line">	-evalue：设置输出结果的e-value值</span><br><span class="line">	-num_descriptions：tabular格式输出结果的条数</span><br><span class="line">	-num_threads：线程数</span><br></pre></td></tr></table></figure>
</li>
</ul>
<p>更多参数信息可直接输入blastn -help查阅。</p>
<ul>
<li><p>Blastn程序的default value为magablast。Megablast 是在intra-species或者very closely related species中使用的blast版本，其特点是word_size较大（默认值为28bp），这样就可以在initial match过程排除掉很多情况，从而加快比对速度。而对于inter-species的两个较短片段就无法搜寻到结果，这种情况可以增加 –task blastn，就一切正常了。</p>
</li>
<li><p>新BLAST+中最拉风的功能是直接控制输出格式，如：-outfmt format “7 qacc sacc evalue length pident”<br>7表示带注释行的tab格式的输出，可以自定义要输出哪些内容，用空格分格跟在7的后面，并把所有的输出控制用双引号括起来，其中qacc查询序列的acc,sacc表示目标序列的acc，evalue即是e值，length即是匹配的长度，pident即是序列相同的百分比。其他可用的选项参见最后。</p>
</li>
<li><p>其它程序，如蛋白序列比对蛋白数据库（blastp）以及核酸序列比对蛋白数据库（blastx）等，用法与blastn类似：</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">blastp -query seq.fasta -out seq.blast -db dbname -outfmt 6 -evalue 1e-5 -num_descriptions 10 -num_threads 8</span><br><span class="line">blastx -query seq.fasta -out seq.blast -db dbname -outfmt 6 -evalue 1e-5 -num_descriptions 10 -num_threads 8</span><br></pre></td></tr></table></figure>
</li>
</ul>
<p>完整的参数说明可以用-help查询。</p>
<h3 id="4-3-也可以直接下载NCBI已经格式化好的库-NCBI-curated-database-，直接在本地库中比对，以蛋白质库nr为例："><a href="#4-3-也可以直接下载NCBI已经格式化好的库-NCBI-curated-database-，直接在本地库中比对，以蛋白质库nr为例：" class="headerlink" title="4.3 也可以直接下载NCBI已经格式化好的库(NCBI-curated database)，直接在本地库中比对，以蛋白质库nr为例："></a>4.3 也可以直接下载NCBI已经格式化好的库(NCBI-curated database)，直接在本地库中比对，以蛋白质库nr为例：</h3><ul>
<li><p>下载库可使用BLAST+自带的update_blastdb.pl脚本</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">cd /opt/blast/blastdb/ 	#切换到本地BLAST库目录</span><br><span class="line">update_blastdb.pl  nr</span><br><span class="line">tar -xzvf *.gz</span><br><span class="line">rm *.gz*  #删除下载的文件</span><br><span class="line">blastp -db nr -query somesequence.fasta</span><br></pre></td></tr></table></figure>
</li>
<li><p>如果没有下载本地化nr库，可添加-remote选项，进行在线比对（就不能使用-num_threads参数了），但速度慢一些，适合序列不多的情况。</p>
</li>
</ul>
<hr>
<blockquote>
<p>Formatting options<br>-outfmt <string><br>alignment view options:<br>0 = Pairwise,<br>1 = Query-anchored showing identities,<br>2 = Query-anchored no identities,<br>3 = Flat query-anchored showing identities,<br>4 = Flat query-anchored no identities,<br>5 = BLAST XML,<br>6 = Tabular,<br>7 = Tabular with comment lines,<br>8 = Seqalign (Text ASN.1),<br>9 = Seqalign (Binary ASN.1),<br>10 = Comma-separated values,<br>11 = BLAST archive (ASN.1),<br>12 = Seqalign (JSON),<br>13 = Multiple-file BLAST JSON,<br>14 = Multiple-file BLAST XML2,<br>15 = Single-file BLAST JSON,<br>16 = Single-file BLAST XML2,<br>17 = Sequence Alignment/Map (SAM),<br>18 = Organism Report<br>Options 6, 7, and 10 can be additionally configured to produce<br>a custom format specified by space delimited format specifiers.<br>The supported format specifiers are:<br>qseqid means Query Seq-id<br>qgi means Query GI<br>qacc means Query accesion<br>qaccver means Query accesion.version<br>qlen means Query sequence length<br>sseqid means Subject Seq-id<br>sallseqid means All subject Seq-id(s), separated by a ‘;’<br>sgi means Subject GI<br>sallgi means All subject GIs<br>sacc means Subject accession<br>saccver means Subject accession.version<br>sallacc means All subject accessions<br>slen means Subject sequence length<br>qstart means Start of alignment in query<br>qend means End of alignment in query<br>sstart means Start of alignment in subject<br>send means End of alignment in subject<br>qseq means Aligned part of query sequence<br>sseq means Aligned part of subject sequence<br>evalue means Expect value<br>bitscore means Bit score<br>score means Raw score<br>length means Alignment length<br>pident means Percentage of identical matches<br>nident means Number of identical matches<br>mismatch means Number of mismatches<br>positive means Number of positive-scoring matches<br>gapopen means Number of gap openings<br>gaps means Total number of gaps<br>ppos means Percentage of positive-scoring matches<br>frames means Query and subject frames separated by a ‘/‘<br>qframe means Query frame<br>sframe means Subject frame<br>btop means Blast traceback operations (BTOP)<br>staxids means unique Subject Taxonomy ID(s), separated by a ‘;’<br>(in numerical order)<br>sscinames means unique Subject Scientific Name(s), separated by a ‘;’<br>scomnames means unique Subject Common Name(s), separated by a ‘;’<br>sblastnames means unique Subject Blast Name(s), separated by a ‘;’<br>(in alphabetical order)<br>sskingdoms means unique Subject Super Kingdom(s), separated by a ‘;’<br>(in alphabetical order)<br>stitle means Subject Title<br>salltitles means All Subject Title(s), separated by a ‘&lt;&gt;’<br>sstrand means Subject Strand<br>qcovs means Query Coverage Per Subject<br>qcovhsp means Query Coverage Per HSP<br>When not provided, the default value is:<br>‘qseqid sseqid pident length mismatch gapopen qstart qend sstart send<br>evalue bitscore’, which is equivalent to the keyword ‘std’<br>Default = `0’</string></p>
</blockquote>
</div><iframe src="/donate/?AliPayQR=/about/AliPayQR.jpg&amp;WeChatQR=/about/wechat_qrcode.jpg&amp;GitHub=null&amp;BTCQR=null&amp;BTCKEY=null&amp;PayPal=null" style="overflow-x:hidden; overflow-y:hidden; border:0xp none #fff; min-height:240px; width:100%;" frameborder="0" scrolling="no"></iframe><div class="tags"><a href="/tags/BLAST/">BLAST</a><a href="/tags/IGV/">IGV</a></div><div class="post-nav"><a class="pre" href="/2017/07/13/mapping-contigs/">Mapping contigs onto reference genome</a><a class="next" href="/2017/07/04/notepad2/">Notepad2代替Windows记事本</a></div></div></div></div><div class="pure-u-1-4 hidden_mid_and_down"><div id="sidebar"><div class="widget"><form class="search-form" action="//www.baidu.com/baidu" method="get" accept-charset="utf-8" target="_blank"><input type="search" name="word" maxlength="20" placeholder="Search"/><input type="hidden" name="si" value="http://blog.ligene.cn"/><input name="tn" type="hidden" value="bds"/><input name="cl" type="hidden" value="3"/><input name="ct" type="hidden" value="2097152"/><input name="s" type="hidden" value="on"/></form></div><div class="widget"><div class="widget-title"><i class="fa fa-folder-o"> 分类</i></div><ul class="category-list"><li class="category-list-item"><a class="category-list-link" href="/categories/Bioinformatics/">Bioinformatics</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/Genomics/">Genomics</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/Linux/">Linux</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/Programming/">Programming</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/Statistics/">Statistics</a></li><li class="category-list-item"><a class="category-list-link" href="/categories/Tutorials/">Tutorials</a></li></ul></div><div class="widget"><div class="widget-title"><i class="fa fa-star-o"> 标签</i></div><div class="tagcloud"><a href="/tags/PubMed/" style="font-size: 15px;">PubMed</a> <a href="/tags/Javascript/" style="font-size: 15px;">Javascript</a> <a href="/tags/IGV/" style="font-size: 15px;">IGV</a> <a href="/tags/C/" style="font-size: 15px;">C</a> <a href="/tags/GitHub/" style="font-size: 15px;">GitHub</a> <a href="/tags/Hexo/" style="font-size: 15px;">Hexo</a> <a href="/tags/Notepad/" style="font-size: 15px;">Notepad</a> <a href="/tags/BLAST/" style="font-size: 15px;">BLAST</a> <a href="/tags/qiime/" style="font-size: 15px;">qiime</a> <a href="/tags/FDR/" style="font-size: 15px;">FDR</a> <a href="/tags/Web/" style="font-size: 15px;">Web</a> <a href="/tags/Microbiome/" style="font-size: 15px;">Microbiome</a> <a href="/tags/Xming/" style="font-size: 15px;">Xming</a> <a href="/tags/PuTTY/" style="font-size: 15px;">PuTTY</a></div></div><div class="widget"><div class="widget-title"><i class="fa fa-file-o"> 最近文章</i></div><ul class="post-list"><li class="post-list-item"><a class="post-list-link" href="/2018/01/25/Xming-PuTTY/">Windows下连接远程Linux桌面</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/01/05/clang/">编程语言之C语言</a></li><li class="post-list-item"><a class="post-list-link" href="/2018/01/01/hello-world/">Hello World</a></li><li class="post-list-item"><a class="post-list-link" href="/2017/10/13/references/">文献检索</a></li><li class="post-list-item"><a class="post-list-link" href="/2017/09/03/qiime-install/">Ubuntu 16.04LTS安装QIIME</a></li><li class="post-list-item"><a class="post-list-link" href="/2017/08/26/multiple-testing/">数据分析的多重检验问题(multiple testing)</a></li><li class="post-list-item"><a class="post-list-link" href="/2017/08/26/binfo-JS/">Javascript-based bioinformatics</a></li><li class="post-list-item"><a class="post-list-link" href="/2017/07/20/Microbiome-visualization/">Microbiome visualization</a></li><li class="post-list-item"><a class="post-list-link" href="/2017/07/13/mapping-contigs/">Mapping contigs onto reference genome</a></li><li class="post-list-item"><a class="post-list-link" href="/2017/07/05/BLAST-Tutorial/">BLAST+使用教程</a></li></ul></div><div class="widget"><div class="widget-title"><i class="fa fa-external-link"> 友情链接</i></div><ul></ul><a href="https://www.haomwei.com/" title="大道至简" target="_blank">大道至简</a><ul></ul><a href="http://blog.shenwei.me/" title="Wei Shen's Note" target="_blank">Wei Shen's Note</a></div></div></div><div class="pure-u-1 pure-u-md-3-4"><div id="footer">Copyright © 2018 <a href="/." rel="nofollow">LIGENE.</a> Powered by<a rel="nofollow" target="_blank" href="https://hexo.io"> Hexo.</a></div></div></div><a class="show" id="rocket" href="#top"></a><script type="text/javascript" src="/js/totop.js?v=0.0.0" async></script><script type="text/javascript" src="//cdn.bootcss.com/fancybox/3.2.5/jquery.fancybox.min.js" async></script><script type="text/javascript" src="/js/fancybox.js?v=0.0.0" async></script><link rel="stylesheet" type="text/css" href="//cdn.bootcss.com/fancybox/3.2.5/jquery.fancybox.min.css"><script type="text/javascript" src="/js/codeblock-resizer.js?v=0.0.0"></script><script type="text/javascript" src="/js/smartresize.js?v=0.0.0"></script></div></body></html>